File: frontend/src/api/followupResponse.ts
import request from '@/utils/request'

export interface FollowupResponseCreate {
  followup_id: number
  overall_feeling?: string
  improvement_level?: string
  medication_adherence?: string
  exercise_volume?: string
  diet_adjustment?: string
  pain_level?: number
  sleep_quality?: string
  daily_activity?: string
  mood_status?: string
  side_effects?: string
  concerns?: string
  additional_info?: string
}

export interface FollowupResponseUpdate {
  overall_feeling?: string
  improvement_level?: string
  medication_adherence?: string
  exercise_volume?: string
  diet_adjustment?: string
  pain_level?: number
  sleep_quality?: string
  daily_activity?: string
  mood_status?: string
  side_effects?: string
  concerns?: string
  additional_info?: string
  is_completed?: boolean
}

export interface FollowupResponseQuery {
  skip?: number
  limit?: number
  patient_id?: number
  followup_id?: number
  is_completed?: boolean
}

export const followupResponseApi = {
  // 创建随访应答
  createResponse: (data: FollowupResponseCreate) => {
    return request.post('/v1/followup-responses/', data)
  },

  // 获取随访应答列表
  getResponses: (params?: FollowupResponseQuery) => {
    return request.get('/v1/followup-responses/', { params })
  },

  // 获取包含患者信息的随访应答列表（机构用户）
  getResponsesWithPatientInfo: (params?: FollowupResponseQuery) => {
    return request.get('/v1/followup-responses/with-patient-info', { params })
  },

  // 获取待回复的随访记录（个人用户）
  getPendingResponses: (params?: { skip?: number; limit?: number }) => {
    return request.get('/v1/followup-responses/pending', { params })
  },

  // 获取单个随访应答
  getResponse: (responseId: number) => {
    return request.get(`/v1/followup-responses/${responseId}`)
  },

  // 更新随访应答
  updateResponse: (responseId: number, data: FollowupResponseUpdate) => {
    return request.put(`/v1/followup-responses/${responseId}`, data)
  },

  // 删除随访应答
  deleteResponse: (responseId: number) => {
    return request.delete(`/v1/followup-responses/${responseId}`)
  },

  // 获取指定患者的随访应答列表
  getPatientResponses: (patientId: number, params?: { skip?: number; limit?: number }) => {
    return request.get(`/v1/followup-responses/patient/${patientId}`, { params })
  }
}

File: backend-python/app/schemas/followup_response.py
from datetime import datetime
from typing import Optional
from pydantic import BaseModel, validator


class FollowupResponseBase(BaseModel):
    """随访应答基础模型"""
    followup_id: int
    overall_feeling: Optional[str] = None  # 总体感受
    improvement_level: Optional[str] = None  # 状况改善
    medication_adherence: Optional[str] = None  # 用药依从性
    exercise_volume: Optional[str] = None  # 运动量
    diet_adjustment: Optional[str] = None  # 饮食调整
    pain_level: Optional[int] = None  # 疼痛程度
    sleep_quality: Optional[str] = None  # 睡眠质量
    daily_activity: Optional[str] = None  # 日常活动能力
    mood_status: Optional[str] = None  # 情绪状态
    social_activity: Optional[str] = None  # 社交活动
    side_effects: Optional[str] = None  # 副作用
    concerns: Optional[str] = None  # 担忧
    suggestions: Optional[str] = None  # 建议
    additional_info: Optional[str] = None  # 其他信息

    @validator('pain_level')
    def validate_pain_level(cls, v):
        if v is not None and (v < 1 or v > 10):
            raise ValueError('疼痛程度必须在1-10之间')
        return v


class FollowupResponseCreate(FollowupResponseBase):
    """创建随访应答模型"""
    is_completed: Optional[bool] = None


class FollowupResponseUpdate(BaseModel):
    """更新随访应答模型"""
    overall_feeling: Optional[str] = None
    improvement_level: Optional[str] = None
    medication_adherence: Optional[str] = None
    exercise_volume: Optional[str] = None
    diet_adjustment: Optional[str] = None
    pain_level: Optional[int] = None
    sleep_quality: Optional[str] = None
    daily_activity: Optional[str] = None
    mood_status: Optional[str] = None
    social_activity: Optional[str] = None
    side_effects: Optional[str] = None
    concerns: Optional[str] = None
    suggestions: Optional[str] = None
    additional_info: Optional[str] = None
    is_completed: Optional[bool] = None

    @validator('pain_level')
    def validate_pain_level(cls, v):
        if v is not None and (v < 1 or v > 10):
            raise ValueError('疼痛程度必须在1-10之间')
        return v


class FollowupResponse(FollowupResponseBase):
    """随访应答响应模型"""
    id: int
    patient_id: int
    response_time: datetime
    is_completed: bool
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class FollowupResponseWithPatient(FollowupResponse):
    """包含患者信息的随访应答模型"""
    patient_name: Optional[str] = None
    patient_phone: Optional[str] = None


class FollowupResponseList(BaseModel):
    """随访应答列表响应模型"""
    responses: list[FollowupResponse]
    total: int
    page: int
    size: int
    pages: int